source/PluginDev/Assets/GooglePlayGames/Platforms/Native/NativeClient.cs

```diff
@@ -285,6 +286,56 @@ private static void LaunchBridgeIntent(IntPtr bridgedIntent)
                 AndroidJNIHelper.DeleteJNIArgArray(objectArray, jArgs);
             }
         }
+        
+        private AndroidJavaObject GetApiClient(GameServices services) {
+            return JavaUtils.JavaObjectFromPointer(C.InternalHooks_GetApiClient(services.AsHandle()));
+        }
+        
+        private string RetrieveUserEmail() {
+            string email;
+            using (AndroidJavaClass jc_plus = new AndroidJavaClass("com.google.android.gms.plus.Plus")) {
+                using (AndroidJavaObject jo_plusAccountApi = jc_plus.GetStatic<AndroidJavaObject>("AccountApi")) {
+                    Debug.Log("jo_plusAccountApi: " + (jo_plusAccountApi == null ? "NULL" : jo_plusAccountApi.ToString()));
+                    using (var apiClient = GetApiClient(mServices)) {
+                        Debug.Log("apiClient: " + (apiClient == null ? "NULL" : apiClient.ToString()));
+                        email  = jo_plusAccountApi.Call<string>("getAccountName", apiClient);
+                        Logger.d("Player email: " + email);
+                    }
+                }
+            }
+            return email;
+        }
+        
+        public string GetToken() {
+			if( String.IsNullOrEmpty(GameInfo.AndroidClientId) || ("__ANDROID_CLIENTID__" == GameInfo.AndroidClientId) )
+			{
+				throw new Exception("Client ID has not been set, cannot request access token.");
+			}
+            string token = null;
+            Debug.Log("Before RetrieveUserEmail");
+            string email = RetrieveUserEmail() ?? "NULL";
+            Debug.Log("After RetrieveUserEmail email: " + email);
+            string scope = "audience:server:client_id:" + GameInfo.AndroidClientId;
```

Comment:
This is a good point. @seaders  - can you rename your GetToken to GetIDToken() - since these are very different tokens, and then we can add the GetAccessToken()?